transfers/amclient.py

```diff
@@ -336,6 +355,80 @@ def download_package(self, uuid):
         else:
             LOGGER.warning('Unable to download package %s', uuid)
 
+    def get_pipelines(self):
+        """GET Archivematica Pipelines (dashboard instances from the storage
+        service.
+        """
+        return utils._call_url_json('{0}/api/v2/pipeline/'.format(self.ss_url),
+                                    params=None,
+                                    headers=self._ss_auth_headers())
+
+    def get_transfer_status(self, transfer_uuid):
+        """GET transfer status if there is a transfer in progress in the
+        Archivematica pipeline. If there aren't an transfers then the
+        response from the server will look as follows:
+
+       ``{"message": "Cannot fetch unitTransfer with UUID
+          ebc8a35c-6742-4264-bc30-22e263966d69", "type": "transfer",
+          "error": true}``
+
+        The response suggesting non-existence is an error, which the caller
+        will have to handle appropriately.
+        """
+        return utils._call_url_json(
+            '{0}/api/transfer/status/{1}/'.format(self.am_url, transfer_uuid),
+            headers=self._am_auth_headers(),
+            params=None)
+
+    def get_processing_config(self, processing_name):
+        """GET a processing configuration file from an Archivematica instance.
+        if the request is succesful an application/xml response is returned
+        to the caller. If the request is unsuccessful then an error code is
+        returned which needs to be handled via error_lookup.
+        """
+        return utils._call_url_json(
+            '{0}/api/processing-configuration/{1}'
+            .format(self.am_url,
+                    processing_name),
+            params=None,
+            headers=self._am_auth_headers())
+
+    def approve_transfer(self, directory, transfer_type=TT_STANDARD):
```

Comment:
In the process of looking at this I did uncover a small bug in approve_transfer so I have completed a re-factor of that function which I'll submit as a separate commit in addressing the other CR points too. It can now use AMClient.